File: pset1/mario.c
#include <stdio.h>
#include <cs50.h>

/*program that recreates this half-pyramid using hashes (#) for blocks.
 first prompt the user for the half-pyramid’s height, a non-negative integer no greater than 23.
 If the user fails to provide a non-negative integer no greater than 23, you should re-prompt for the same again.
*/

int ask(){
    printf("Height: ");
    int height = get_int();
    return height;
}

int check(height)
{
    if (height >= 0 && height <= 23)
    {   
        return height;
    }
    else
    {
        return -1;  // -1 acts as a flag
    }
}



int main(void){
    int h = ask(); // asks and gets the input number for the height
    int result = check(h);  // 
    while (result == -1)
    {
        h = ask();
        result = check(h);
    }
    

    for(int l = 0; l<h;l++)
    {
        for(int b = h ;b>1+l;b--)  
        {
            printf(" ");
        }
        for(int c = 0;c<1+l;c++) 
        {
            printf("#");
        }
        for( int m = 0; m< 2;m++)
        {
            printf(" ");
        }
        for(int d = 0;d<l+1;d++) 
        {
            printf("#");
        }
        printf("\n");
    }
}

File: pset6/mario.py
#Implement a program that prints out a half-pyramid of a specified height, per the below.
#
#$ ./mario
#Height: 5
#     ##
#    ###
#   ####
#  #####
# ######

#1. prompt the user and validade input
#draw half pyramid

import cs50

def main():
    n = get_positive_int()
    #draw half pyramid: spaces, #, new line
    b = n-1
    h = 2
    for i in range(n):
        print(" "*b, end = "")
        print("#"*h) #prints # + new line
        b=b-1
        h=h+1

def get_positive_int():
    while True:
        print("Height: ", end="")
        n = cs50.get_int()
        if n >= 0:
            break
    return n

if __name__ == "__main__":
    main()
    
